### DISCUSSION TITLE
Button to Download All Images in a post

### DISCUSSION THREAD

**Discussion by drainz7** [HOAIAN2/Instagram-Downloader#4] [Ideas]
This extension work great for me, other extension get me random cropped images, but not this app, for me it would be more help if there is a button to download all fetched images, because lot post have 10 images in it, so we don't need to click 1 by 1, or maybe an option for when we click the download button it automatically download all images fetched in a post, thank you for your awesome work

**Comment by hoaianle** [comment]
It Ok, but you need to allow www.instagram.com download multi files because Chrome detect it spam. So should I build that feature?

**Reply by drainz7** [reply]
Yes please, it would so much help for me, Thank you so much

**Reply by hoaianle** [reply]
Try this, click on ```Photos``` button to download all. I willl build it later but you can paste it to the end of ```handleEvents()``` function in main.js file.
```js
const button = document.querySelector('.title-container>span')
button.addEventListener('click', ()=> {
    document.querySelectorAll('.photos-items').forEach(img=>{
        img.click()
    })
})
```

**Reply by hoaianle** [reply]
But you shouldn't click fast because it laggy so much. That why I never do that before.

**Comment by drainz7** [comment]
Thank you, it work perfect for me, you are awesome! 🥇 🥇 🥇

**Reply by hoaianle** [reply]
Hey, I just push new version ( didn't release yet but you can clone git repo). It support download multiple files as zip and mark which one saved. I will update demo in README and release. Thanks for using my Extension.

**Reply by drainz7** [reply]
Thank You So Much! You're the best 👍 👍 👍

**Comment by hridaydas** [comment]
The app not working with a private profile I followed. Am I missing something here? Thanks.

**Reply by hoaianle** [reply]
Something missing on token, I will check later.

**Reply by hoaianle** [reply]
> The app not working with a private profile I followed. Am I missing something here? Thanks.

OK, I just release new version [Instagram Downloader v3.0.1](https://github.com/HOAIAN2/Instagram-Downloader/releases/tag/v3.0.1). Now you can download private profile ( I test it with my friend), it will took you more around 200-500ms , if you have any issue comment bellow.

**Reply by hridaydas** [reply]
Sure, thanks. 

![ii](https://github.com/HOAIAN2/Instagram-Downloader/assets/6022568/46b24e3d-2b0c-41da-9d27-c93524a81e11)
`

**Reply by hoaianle** [reply]
Update to 3.0.1 version, I just fix that reading '0' bug
